src/EventStore.Projections.Core/Services/Processing/ResultEventEmitter.cs

```diff
@@ -20,21 +23,31 @@ public class ResultEventEmitter : IResultEventEmitter {
 			CheckpointTag at) {
 			var streamId = _namesBuilder.MakePartitionResultStreamName(partition);
 			var allResultsStreamId = _namesBuilder.GetResultStreamName();
+			EmittedEventEnvelope result;
 			if (string.IsNullOrEmpty(partition)) {
-				var result =
+				result =
 					new EmittedEventEnvelope(
 						projectionResult == null
 							? new EmittedDataEvent(
 								streamId, Guid.NewGuid(), "ResultRemoved", true, null, null, at, null)
 							: new EmittedDataEvent(
 								streamId, Guid.NewGuid(), "Result", true, projectionResult, null, at, null),
 						_resultStreamMetadata);
+				if (!string.IsNullOrEmpty(result.Event.Data) && !warningProjectionList.Contains(_namesBuilder.EffectiveProjectionName)) {
+					if (System.Text.ASCIIEncoding.Unicode.GetByteCount(result.Event.Data) / Math.Pow(10, 6) >= 8) {
+						Log.Warning(
+							"State size for the Projection {p} is greater than 8 MB. State size for a projection should be less than 16 MB. Current state size for Projection {p} is {s} MB",
+							_namesBuilder.EffectiveProjectionName, _namesBuilder.EffectiveProjectionName,
+							System.Text.ASCIIEncoding.Unicode.GetByteCount(result.Event.Data) / Math.Pow(10, 6));
+					}
+					warningProjectionList.Add(_namesBuilder.EffectiveProjectionName);
```

Comment:
isn't this going to add the projection to the list even if the state is small, and then when the state is big it won't log anything?

Reply:
Yeah it was a mistake on my end, rectified.